### ISSUE TITLE
Accessibility: LocalizedControlType property is not publicly accessible for a DataGridViewCellAccessibleObject

### ISSUE THREAD

**Issue by Olina-Zhang** [dotnet/winforms#5792]
* .NET Core Version: 
.Net SDK 6.0.100-rc.2.21466.17

* Have you experienced this same bug with .NET Framework?: 
Yes

**Problem description:**
Related AzDo bug: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1404297.
 For a specific case in the Settings designer with a **custom** DataGridViewCell **where the pane containing the edit and button control** is missing the LocalizedControlType.

**Actual behavior:**
AccessibilityInsight detectes an error about "the localizedControlType property must not be null", and in inspect, it is missing ControlType and LocalizedControlType properies:
![image](https://user-images.githubusercontent.com/26474449/133731662-971dae6f-8cff-4a2d-b344-34de16985e41.png)
![image](https://user-images.githubusercontent.com/26474449/133732057-93feeea2-1df4-4379-85d0-276e894a5143.png)

**Expected behavior:**
No error
